### PR TITLE
version: support post versions

### PR DESCRIPTION
**Pull request by dtrodrigues**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?
- [x] Have you successfully run `brew man` locally and committed any changes?

-----
Add support for PEP 440 post versions. Relates to https://github.com/Homebrew/brew/pull/8832. In support of https://github.com/Homebrew/homebrew-core/pull/61861 so the version number does not need to be manually specified.

### CODE REVIEW

**Comment by dtrodrigues** [comment]
Not sure what to do here. This PR makes the explicit version line in `legit.rb` (https://github.com/Homebrew/homebrew-core/blob/092220c889944cd078f6a7416f9580b19e34e699/Formula/legit.rb#L7) redundant and causes testing on this change to fail, but changing that line in `legit` before another tagged brew version probably wouldn't work either.

**Review by MikeMcQuaid — APPROVED** [review]
👍🏻 once CI is green.

**Comment by MikeMcQuaid** [comment]
> Not sure what to do here. This PR makes the explicit version line in `legit.rb` (https://github.com/Homebrew/homebrew-core/blob/092220c889944cd078f6a7416f9580b19e34e699/Formula/legit.rb#L7) redundant and causes testing on this change to fail, but changing that line in `legit` before another tagged brew version probably wouldn't work either.

I'm not sure what best to do, either. Could just override the required step here and merge the two PRs in quick succession before a tag?

**Comment by reitermarkus** [comment]
> Not sure what to do here.

Add an exception for the formula to the cop and remove it once it is fixed?

**Comment by dtrodrigues** [comment]
I submitted a PR for `legit` (https://github.com/Homebrew/homebrew-core/pull/62121). Merging both quickly and then tagging should work, but is a little risky. A temporary allowlist for redundant versions until the next tag should work, but the challenge is trying to balance doing it "the right way" without requiring too much throwaway work.

**Comment by reitermarkus** [comment]
Why is a new tag necessary? `homebrew-core` CI runs on `master` anyways, right?

**Comment by MikeMcQuaid** [comment]
> Why is a new tag necessary? `homebrew-core` CI runs on `master` anyways, right?

@reitermarkus It'll break installation for anyone installing not yet on `master`.

**Comment by MikeMcQuaid** [comment]
Thanks @dtrodrigues!

File: Library/Homebrew/dev-cmd/audit.rb [hunk 1]
```diff
@@ -1024,7 +1024,7 @@ def audit_version
       elsif !version.detected_from_url?
         version_text = version
         version_url = Version.detect(url, **specs)
-        if version_url.to_s == version_text.to_s && version.instance_of?(Version)
+        if version_url.to_s == version_text.to_s && version.instance_of?(Version) && @name != "legit"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/audit.rb:1027]: ` && @name != "legit"` can now be removed because I've just tagged 2.5.3.

**Pull request merged**